kvirt/cluster/k3s/ctlplanes.sh

```diff
@@ -9,4 +9,4 @@ apt-get -y install curl
 echo bpffs /sys/fs/bpf bpf defaults 0 0 >> /etc/fstab
 mount /sys/fs/bpf
 {% endif %}
-curl -sfL https://get.k3s.io | {{ install_k3s_args|default("") }} K3S_TOKEN={{ token }} sh -s - server --server https://{{ first_ip }}:6443 {{ extra_args|join(" ") }}
+curl -sfL https://get.k3s.io | {{ install_k3s_args|default("") }} K3S_TOKEN={{ token }} sh -s - server --server https://{{ first_ip }}:6443 {{ extra_args|join(" ") }} --tls-san {{ first_ip }}
```

Comment:
It's unclear what this snippet does

Reply:
I believe this is not needed

Reply:
It simply ensure that the `--tls-san` parameter is used. And I'm pretty sure one should use the same parameters across all `K3s server nodes`. Except for the `--cluster-init` parameter as far as I know.

If you need it, I can ask in the K3s RancherLabs Slack community for some info on whether it's needed and best practice to add the `--tls-san` parameter to all `k3s server nods` or not.

Reply:
please do so. My understanding is that the tls-san ip indicate additional subject names to add to the certificates, which should only be needed when bootstrapping.
No need to repeat the data in the joining phase

Reply:
I've reached out to them. Waiting for an response.

Reply:
Here's the response.

![image](https://github.com/karmab/kcli/assets/80959443/905bd6e1-ebfd-441e-98ed-882481ac5c3f)

---

So as I understand it. `--tls-san` on all `K3s server nodes` in all cases. E.g. `node count > 1` and `DNS name` for the `API IP` and so on.

Thanks and what do you think @karmab

Reply:
My reading of this is that this setting is only needed in the bootstrap call so that the generated certificates include the vip/fqdn as a part of the subject, but those certs are shared between all the control plane nodes.
so 
- it's correct to include the fqdn instead of a vip (in cloud cases)
- no changes is required in joining phases (regardless of whether the node is an extra control plane or a regular worker node)

Reply:
What about

![image](https://github.com/karmab/kcli/assets/80959443/57b399da-ff6c-4b2e-b39b-31753ca96a76)

---

I understand that as this being necessary on all servers. It is, of course, available via `KCLI` to set such extra arguments via **KCLI's** `extra_ctlplane_args` option in the **KCLI** parameter file. So this back and forth is more about where it's best to "put" this ... as we often have these discussions in regards to where it's logical to "put" logic when using **KCLI** 🥇 

---

Let me know what you think.

Reply:
if the vip entry was really required on all nodes, then current deployment would be failing.

Reply:
I agree with that. However, I've already described how I understand Mr. Brandon David. Do you also understand it that way? I mean. What we're trying to achieve is that the K3s installation is secure - this specifically coming from the need to patch this [GHSA-m4hf-6vgr-75r2](https://github.com/k3s-io/k3s/security/advisories/GHSA-m4hf-6vgr-75r2) CVE.

And I understand it - `--tls-san` is needed on all server nodes as is what Mr. Brandon David is also saying.